tabular/src/autogluon/tabular/models/imodels/imodels_models.py

```diff
@@ -0,0 +1,169 @@
+from abc import abstractmethod
+
+import numpy as np
+import pandas as pd
+from autogluon.core.models import AbstractModel
+from autogluon.core.utils.try_import import try_import_imodels
+from autogluon.features.generators import LabelEncoderFeatureGenerator
+from autogluon.tabular.models.lr.lr_preprocessing_utils import OheFeaturesGenerator
+from autogluon.core.features.types import R_INT, R_FLOAT, R_CATEGORY, R_OBJECT
+from autogluon.tabular.models.lr.hyperparameters.parameters import get_param_baseline, INCLUDE, IGNORE, ONLY, _get_solver, preprocess_params_set
+
+class IModelsModel(AbstractModel):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._feature_generator = None
+
+    @abstractmethod
+    def get_model(self):
+        return NotImplemented
+
+    def _preprocess(self, X: pd.DataFrame, is_train=False, **kwargs) -> np.ndarray:
+        X = super()._preprocess(X, **kwargs)
+
+        if is_train:
+            categorical_featnames = self._get_types_of_features(X)
+            self._feature_generator = OheFeaturesGenerator(cats_cols=categorical_featnames['categorical']) # LabelEncoderFeatureGenerator(verbosity=0)
+            self._feature_generator.fit(X=X)
```

Comment:
Why use `OheFeaturesGenerator` specifically? I probably wouldn't recommend it as this is an old class that should probably be reworked. To my knowledge this will not function the way you intend, as it will drop any non-categorical features. For now, add a simple sklearn OHE logic:

```
from sklearn.preprocessing import OneHotEncoder
```

```
X = super()._preprocess(X, **kwargs)
if is_train:
    self._categorical_featnames = self._feature_metadata.get_features(valid_raw_types=['category'])
    self._other_featnames = self._feature_metadata.get_features(invalid_raw_types=['category'])
    if self._categorical_featnames:
        self._ohe = OneHotEncoder(dtype=np.uint8, handle_unknown='ignore')
        self._ohe.fit(X=X[self._categorical_featnames])
        self._ohe_columns = self._ohe.get_feature_names_out()

if self._ohe is not None:
    X_index = X.index
    X_ohe = self._ohe.transform(X[self._categorical_featnames])
    X_ohe = pd.DataFrame.sparse.from_spmatrix(X_ohe, columns=self._ohe_columns, index=X_index)
    if self._other_featnames:
        X = pd.concat([X[self._other_featnames], X_ohe], axis=1)
    else:
        X = X_ohe

return X.fillna(0)
```

This should be significantly faster (4x+).

Reply:
Switched over to this!